python-frontend/src/main/java/org/sonar/python/semantic/v2/types/TrivialTypeInferenceVisitor.java

```diff
@@ -451,10 +453,32 @@ public void visitAssignmentStatement(AssignmentStatement assignmentStatement) {
     getFirstAssignmentName(assignmentStatement).ifPresent(lhsName -> {
       var assignedValueType = assignmentStatement.assignedValue().typeV2();
       lhsName.typeV2(assignedValueType);
-      addStaticFieldToClass(lhsName);
+      addStaticFieldToClass(lhsName, PythonType.UNKNOWN);
     });
   }
 
+  @Override
+  public void visitAnnotatedAssignment(AnnotatedAssignment assignmentStatement) {
+    scan(assignmentStatement.assignedValue());
+    scan(assignmentStatement.annotation());
+    scan(assignmentStatement.variable());
+
+    Optional.ofNullable(assignmentStatement.variable())
+      .filter(NameImpl.class::isInstance)
+      .map(NameImpl.class::cast)
+      .ifPresent(lhsName -> {
+        if (currentType() instanceof ClassType) {
```

Comment:
_question_ this line is to check that you are in a class definition right?

Reply:
discussed